container-storage-setup.sh

```diff
@@ -727,8 +727,18 @@ grow_root_pvs() {
   # partitions on all disks, so as long as they match, growing the LV should
   # also work.
   for pv in $_ROOT_PVS; do
-    # Split device & partition.  Ick.
-    growpart $( echo $pv | sed -r 's/([^0-9]*)([0-9]+)/\1 \2/' ) || true
+    if ! test -b $pv; then
+        Error "Not a block device: $pv"
+    fi
+    local major_hex=$(stat -c '%t' $pv)
+    local minor_hex=$(stat -c '%T' $pv)
+    local major_minor=$((0x${major_hex})):$((0x${minor_hex}))
+    local devpath=$(realpath /sys/dev/block/$major_minor)
+    local partition=$(cat $devpath/partition)
```

Comment:
Should the script abort if any of these commands fail, such as this one if the device is not a partition?  The `set -e` won't catch this since bash interprets the `local` statement as a successful simple command, so the exit status of the command substitution is ignored.